tests/e2e/multicluster/multicluster_multiprimary_test.go

```diff
@@ -0,0 +1,479 @@
+	Expect(kubectl.CreateNamespace(ns, kubeconfig2)).To(Succeed(), "Namespace failed to be created")
+
+	// Label the namespace
+	Expect(kubectl.Patch("", "namespace", ns, "merge", `{"metadata":{"labels":{"istio-injection":"enabled"}}}`)).
+		To(Succeed(), "Error patching sample namespace")
+	Expect(kubectl.Patch("", "namespace", ns, "merge", `{"metadata":{"labels":{"istio-injection":"enabled"}}}`, kubeconfig2)).
+		To(Succeed(), "Error patching sample namespace")
+
+	version := istioVersion.Version
+	// Deploy the sample app from upstream URL in both clusters
+	if istioVersion.Name == "latest" {
+		version = "master"
+	}
+	helloWorldURL := fmt.Sprintf("https://raw.githubusercontent.com/istio/istio/%s/samples/helloworld/helloworld.yaml", version)
+	Expect(kubectl.ApplyWithLabels(ns, helloWorldURL, "service=helloworld", kubeconfig)).To(Succeed(), "Sample service deploy failed on Primary Cluster")
+	Expect(kubectl.ApplyWithLabels(ns, helloWorldURL, "service=helloworld", kubeconfig2)).To(Succeed(), "Sample service deploy failed on Remote Cluster")
+
+	Expect(kubectl.ApplyWithLabels(ns, helloWorldURL, "version=v1", kubeconfig)).To(Succeed(), "Sample service deploy failed on Primary Cluster")
+	Expect(kubectl.ApplyWithLabels(ns, helloWorldURL, "version=v2", kubeconfig2)).To(Succeed(), "Sample service deploy failed on Remote Cluster")
+
+	sleepURL := fmt.Sprintf("https://raw.githubusercontent.com/istio/istio/%s/samples/sleep/sleep.yaml", version)
+	Expect(kubectl.Apply(ns, sleepURL, kubeconfig)).To(Succeed(), "Sample sleep deploy failed on Primary Cluster")
+	Expect(kubectl.Apply(ns, sleepURL, kubeconfig2)).To(Succeed(), "Sample sleep deploy failed on Remote Cluster")
+}
+
+// getPodName returns the pod name of the given deployment in the given namespace
+func getSleepPodName(ctx context.Context, cl client.Client, ns, deploymentName string) string {
+	samplePods := &corev1.PodList{}
+	cl.List(ctx, samplePods, client.InNamespace(ns))
+	var sleepPodName string
+	for _, pod := range samplePods.Items {
+		if pod.Labels["app"] == deploymentName {
+			sleepPodName = pod.GetName()
+			break
+		}
+	}
+
+	return sleepPodName
+}
+
+// getListCurlResponses runs the curl command 10 times from the sleep pod in the given cluster and get response list
+func getListCurlResponses(ns, podName, kubeconfig string) []string {
+	var responses []string
+	for i := 0; i < 10; i++ {
+		response, err := kubectl.Exec(ns, podName, "sleep", "curl -sS helloworld.sample:5000/hello", kubeconfig)
+		Expect(err).NotTo(HaveOccurred())
+		responses = append(responses, response)
+	}
+	return responses
+}
+
+// getSVCAddress returns the address of the istiod service in the given cluster
+func getSVCAddress(ctx context.Context, cl client.Client, ns string) string {
+	svc := &corev1.Service{}
+	err := cl.Get(ctx, kube.Key("istio-eastwestgateway", ns), svc)
+	Expect(err).NotTo(HaveOccurred())
+	return svc.Status.LoadBalancer.Ingress[0].IP
```

Comment:
This will panic if Ingress isn't ready which could make this failure more confusing. Can you first add a check to ensure that ingress has at least one element?

Reply:
Added a validation to wait until the Loadbalancer to be ready